oslc4j-core/src/main/java/org/eclipse/lyo/oslc4j/core/model/ServiceProviderFactory.java

```diff
@@ -354,12 +354,18 @@ private static String resolvePathParameters(final String basePath, final String
 
 		String returnUri = null;
 		
+		Map<String, Object> parameters = pathParameterValues;
+		if (parameters == null) {
+			parameters = new HashMap<>();
```

Comment:
I don't see the motivation for this change.
It seems that you are trying to handle the case where the input parameter pathParameterValues is null, since the caller does not have path parameters.
Fine, but shouldn't the caller simple send `new HashMap<>()` instead of null?

This way, we don't have to take the risk of changing this method in core. and I believe this makes sure the caller know what they are doing.